torch/testing/_internal/common_utils.py

```diff
@@ -331,6 +331,7 @@ def _check_module_exists(name):
 TEST_SCIPY = _check_module_exists('scipy')
 TEST_MKL = torch.backends.mkl.is_available()
 TEST_NUMBA = _check_module_exists('numba')
+TEST_WITH_ROCM = (torch.version.hip is not None) and torch.cuda.is_available()
```

Comment:
Nit.  Really I don't care either way, but read on:

I think we've set a precedent elsewhere to write checks like `torch.version.hip is not None`, but it would seem more pythonic to drop these `is None` or `is not None` checks.  The way it has worked for as long as I can remember, `torch.version.hip` will be None when PyTorch is built for CUDA, and vice versa for `torch.version.cuda` when compiling for ROCm/HIP.

At the very least, do we need the parens?  Was that for clarity?

Or, we could go so far as to use
```suggestion
TEST_WITH_ROCM = torch.version.hip and torch.cuda.is_available()
```

Reply:
Yeah, I like parens too much. I'm actually also somewhat of a fan of testing Optional[String] for `is/is not None` when checking whether the Optional is present, but that might be me, too...